VideoApi/VideoApi.Common/Security/QuickLinksJwtTokenProvider.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using VideoApi.Common.Configuration;
+using VideoApi.Domain.Enums;
+
+namespace VideoApi.Common.Security
+{
+    public class QuickLinksJwtTokenProvider : IQuickLinksJwtTokenProvider
+    {
+        private readonly QuickLinksConfiguration _quickLinksConfiguration;
+        public const int ExpiryTimeInMinutes = 60 * 8;
+
+        public QuickLinksJwtTokenProvider(IOptions<QuickLinksConfiguration> quickLinksConfiguration)
+        {
+            _quickLinksConfiguration = quickLinksConfiguration.Value;
+        }
+
+        public QuickLinksJwtDetails GenerateToken(string name, string userName, UserRole role)
+        {
+            var key = Convert.FromBase64String(_quickLinksConfiguration.JwtProviderSecret);
+
+            var claims = new ClaimsIdentity(new[]
+            {
+                new Claim(ClaimTypes.Name, name),
+                new Claim(ClaimTypes.GivenName, name),
+                new Claim(ClaimTypes.Surname, name),
+                new Claim("preferred_username", userName),
+                new Claim("name", name), // Display Name
```

Comment:
How are these strings used?

Reply:
that comes in the token as we implemented for v2.0